core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -199,6 +210,25 @@ protected void doStop() throws Exception {
     ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
     context.setContextPath("/");
     context.addServlet(servletHolder, "/*");
+    String authMethod = getConfiguration().getString(RestConfig.AUTHENTICATION_METHOD);
+    if (!"NONE".equals(authMethod)) {
+      String realm = getConfiguration().getString(RestConfig.AUTHENTICATION_REALM);
+      List<String> roles = getConfiguration().getList(RestConfig.AUTHENTICATION_ROLES);
+      final ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
+      Constraint constraint = new Constraint();
+      constraint.setAuthenticate(true);
+      constraint.setRoles(roles.toArray(new String[roles.size()]));
+      ConstraintMapping constraintMapping = new ConstraintMapping();
+      constraintMapping.setConstraint(constraint);
+      constraintMapping.setMethod("*");
+      constraintMapping.setPathSpec("/*");
+      securityHandler.addConstraintMapping(constraintMapping);
```

Comment:
the way rest-utils does programatic configuration means we don't get the benefit(?) of the jetty xml configability: http://www.eclipse.org/jetty/documentation/9.4.x/configuring-security-authentication.html#_scoping_security_realms

to what @alexlod said, i was thinking of this as barebones what we need now and once we need per-path auth we would fight that battle